app/controllers/nonsipp/landorpropertydisposal/LandPropertyDisposalCYAController.scala

```diff
@@ -0,0 +1,640 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.landorpropertydisposal
+
+import cats.implicits.toShow
+import config.Refined.{Max50, Max5000}
+import controllers.PSRController
+import controllers.actions._
+import controllers.nonsipp.landorpropertydisposal.LandPropertyDisposalCYAController._
+import models.HowDisposed.{HowDisposed, Other, Sold, Transferred}
+import models.SchemeId.Srn
+import models._
+import navigation.Navigator
+import pages.nonsipp.landorproperty.LandOrPropertyAddressLookupPage
+import pages.nonsipp.landorpropertydisposal._
+import play.api.i18n._
+import play.api.mvc._
+import utils.DateTimeUtils.localDateShow
+import utils.ListUtils.ListOps
+import viewmodels.DisplayMessage._
+import viewmodels.implicits._
+import viewmodels.models._
+import views.html.CheckYourAnswersView
+
+import java.time.LocalDate
+import javax.inject.{Inject, Named}
+
+class LandPropertyDisposalCYAController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  view: CheckYourAnswersView
+) extends PSRController {
+
+  def onPageLoad(
+    srn: Srn,
+    index: Max5000,
+    disposalIndex: Max50,
+//    page: Int,
+    checkOrChange: CheckOrChange
+  ): Action[AnyContent] =
+    identifyAndRequireData(srn) { implicit request =>
+      (
+        for {
+
+          howWasPropertyDisposed <- requiredPage(HowWasPropertyDisposedOfPage(srn, index, disposalIndex))
+          addressLookUpPage <- requiredPage(LandOrPropertyAddressLookupPage(srn, index))
+          landOrPropertyStillHeld <- requiredPage(LandOrPropertyStillHeldPage(srn, index, disposalIndex))
```

Comment:
@narendrapaduchuri when I use the test endpoints (test-only) to create some data, code breaks somewhere around these lines.

Reply:
@narendrapaduchuri this is still not working, but seems the test-generator is not creating all the data. So I think it is an independent issue from the CYA page